projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/__tests__/deploy-job-component.spec.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023-2023 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import DeployJobDialog, {
+  IDeployJobDialogProps
+} from '../components/DeployJob';
+import { render, fireEvent } from '@testing-library/react';
+import { jobData } from '../jobData';
+import { VdkOption } from '../vdkOptions/vdk_options';
+
+const defaultProps: IDeployJobDialogProps = {
+  jobName: 'test-name',
+  jobTeam: 'test-team',
+  jobPath: 'test-path'
+};
+
+describe('#constructor()', () => {
+  it('should return a new instance', () => {
+    const box = new DeployJobDialog(defaultProps);
+    expect(box).toBeInstanceOf(DeployJobDialog);
+  });
+});
+
+describe('#render()', () => {
+  it('should return contain job name input with placeholder equal to jobName from props', () => {
+    const box = new DeployJobDialog(defaultProps);
+    const component = render(box.render());
+    const input = component.getByPlaceholderText(defaultProps.jobName);
+    expect(input).toBe(component.getAllByLabelText('Job Name:')[0]);
```

Comment:
This tests only check how the component is loaded - since we will be directly fetching job name from config.ini when the component is loaded the placeholder for the Job Name should include that name. This tests is for that.

Reply:
About the input validation: currently only simple checks are done in the UI since more checks are done in the server. If there is a problem with the input value it will be written in the server response